readme.md

```diff
@@ -339,27 +373,72 @@ Create a key with a default value.
 
 The default value is written to the actual `UserDefaults` and can be used elsewhere. For example, with a Interface Builder binding.
 
-#### `Defaults.NSSecureCodingKey` *(alias `Defaults.Keys.NSSecureCodingKey`)*
+#### `Defaults.Serializable`
 
 ```swift
-Defaults.NSSecureCodingKey<T>(_ key: String, default: T, suite: UserDefaults = .standard)
+public protocol DefaultsSerializable {
+	static var bridge: Defaults.Bridge { get }
+}
 ```
 
-Type: `class`
+Type: `protocol`
 
-Create a NSSecureCoding key with a default value.
+A protocol which can do serialization and de-serialization easily.
 
-The default value is written to the actual `UserDefaults` and can be used elsewhere. For example, with a Interface Builder binding.
+It should have a static variable `bridge` and its protocol should conform to `Defaults.Bridge`.
 
-#### `Defaults.NSSecureCodingOptionalKey` *(alias `Defaults.Keys.NSSecureCodingOptionalKey`)*
+#### `Defaults.CollectionSerializable`
 
 ```swift
-Defaults.NSSecureCodingOptionalKey<T>(_ key: String, suite: UserDefaults = .standard)
+public protocol DefaultsCollectionSerializable: Collection, Defaults.Serializable {
+	init(_ elements: [Element])
+}
 ```
 
-Type: `class`
+Type: `protocol`
+
+A `Collection` which can store into the `UserDefaults`.
+
+It should have an initializer `init(_ elements: [Element])` to let `Defaults` do the de-serialization.
+
+#### `Defaults.SetAlgebraSerializable`
+
+```swift
+public protocol DefaultsSetAlgebraSerializable: SetAlgebra, Defaults.Serializable {
+	func toArray() -> [Element]
+}
+```
+
+Type: `protocol`
+
+A `SetAlgebra` which can store into the `UserDefaults`.
+
+It should have a function `func toArray() -> [Element]` to let `Defaults` do the serialization.
+
+#### `Defaults.Bridge`
+
+```swift
+public protocol DefaultsBridge {
+	associatedtype Value
+	associatedtype Serializable
+	func serialize(_ value: Value?) -> Serializable?
+	func deserialize(_ object: Serializable?) -> Value?
+}
+```
+
+Type: `protocol`
+
+A Bridge can do the serialization and de-serialization.
+
+Have two associate types `Value` and `Serializable`.
 
-Create a NSSecureCoding key with an optional value.
+`Value` is the type user want to use it.
+
+`Serializable` is the type stored in `UserDefaults`.
+
+`serialize` will be executed before storing to the `UserDefaults` .
+
+`deserialize` will be executed after retrieving its value from the `UserDefaults`.
```

Comment:
I haven't checked, but make sure that all the docs for the types here are also in the source code as doc comments for the types, including examples.